AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_component/physx_collider.py

```diff
@@ -0,0 +1,318 @@
+
+    # Shape: Cylinder
+    def set_cylinder_shape(self) -> None:
+        """
+        Property Type, Default Visibility - ('EditorProxyCylinderShapeConfig', 'NotVisible')
+
+        Used to set the Physx Collider's Shape property to Cylinder.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_Cylinder)
+
+    def set_cylinder_subdivision(self, subdivisions: int) -> None:
+        """
+        Property Type, Default Visibility - ('unsigned char', 'Visible')
+
+        Used to set the Physx Collider's Cylinder Shape's subdivision. Subdivision supports int values [3-125].
+        """
+        assert self.component.is_property_visible(self.Path.Cylinder.SUBDIVISION), \
+            f"Failure: Cannot set cylinder subdivisions when property is not visible. Set the shape to cylinder first."
+
+        self.component.set_component_property_value(self.Path.Cylinder.SUBDIVISION, subdivisions)
+
+    def set_cylinder_height(self, height: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Cylinder Shape's height.
+        """
+        assert self.component.is_property_visible(self.Path.Cylinder.HEIGHT), \
+            f"Failure: Cannot set cylinder height when property is not visible. Set the shape to cylinder first."
+
+        self.component.set_component_property_value(self.Path.Cylinder.HEIGHT, float(height))
+
+    def set_cylinder_radius(self, radius: float) -> None:
+        """
+        Property Type, Default Visibility - ('float', 'Visible')
+
+        Used to set the Physx Collider's Cylinder Shape's radius.
+        """
+        assert self.component.is_property_visible(self.Path.Cylinder.RADIUS), \
+            f"Failure: Cannot set cylinder radius when property is not visible. Set the shape to cylinder first."
+
+        self.component.set_component_property_value(self.Path.Cylinder.RADIUS, float(radius))
+
+    # Shape: PhysicsAsset
+    def set_physicsasset_shape(self) -> None:
+        """
+        Property Type, Default Visibility - ('PhysicsAssetShapeConfiguration', 'ShowChildrenOnly')
+
+        Used to set the Physx Collider's Shape property to PhysicsAsset.
+        """
+        self.component.set_component_property_value(self.Path.SHAPE, physics.ShapeType_PhysicsAsset)
+
+    def set_physx_mesh_from_path(self, asset_product_path: str) -> None:
+        """
+        Property Type, Default Visibility - ('Asset<MeshAsset>', 'Visible')
+
+        Used to set the Physx Collider's PhysicsAsset Shape's Physx Mesh from a user provided asset product path.
+            This will return an error if a source path is provided.
+        """
+        assert self.component.is_property_visible(self.Path.PhysicsAsset.PHYSX_MESH), \
+            f"Failure: Cannot set Physics Mesh when property is not visible."
+
+        px_asset = Asset.find_asset_by_path(asset_product_path)
```

Comment:
Does `find_asset_by_path` fails internally if the asset was not found? If not, then maybe it should error here.

Reply:
`find_asset_by_path(...)` queries the Asset Catalog Request Bus and gets an `azlmbr.asset.asset_id` object from the bus and then will assert if the asset_id is not valid or the `is_valid()` call cannot be completed.